apps/api/pages/api/bookings/_get.ts

```diff
@@ -162,6 +162,10 @@ function buildWhereClause(
 
 async function handler(req: NextApiRequest) {
   const { userId, isAdmin, prisma } = req;
+
+  const dateFrom = req.query.dateFrom ? new Date(req.query.dateFrom as string) : undefined;
+  const dateTo = req.query.dateTo ? new Date(req.query.dateTo as string) : undefined;
+
```

Comment:
I wonder if we can do something like this:
```suggestion
const args: Prisma.BookingFindManyArgs = {
  include: {
    attendees: true,
    user: true,
    payment: true,
  },
  where: {}, // Initialize 'where' here
};
```
from current L169 to L174
That would initialize where inside args and then we could have 
```
if (dateFrom) {
  args.where.startTime = { gte: dateFrom };
}
if (dateTo) {
  args.where.endTime = { lte: dateTo };
}
```
later on without the issue you mention

Reply:
Hey, thx for the feedback the support here is killer!

Unfortunately, even when initializing args with `where`, I still get a type issue: `'args.where' is possibly 'undefined'.`

<img width="458" alt="Screenshot 2023-12-20 at 5 46 11 PM" src="https://github.com/calcom/cal.com/assets/36115192/2a0ca716-7356-4a84-b9b8-71f013442752">

This is because the `BookingFindManyArgs` type sets `where` as optional and args is cast as `BookingFindManyArgs`.

<img width="496" alt="Screenshot 2023-12-20 at 5 46 38 PM" src="https://github.com/calcom/cal.com/assets/36115192/42906bc6-e6cf-4d4c-b018-a0fdbd18abe9">

There are two ways around this: 
- update the type
- re-cast the type

However, I want this PR to be as minimal as possible as I don't have context on any downstream implications or any new potential errors introduced by an empty `where` object `{}`.

For these reasons, I recommend sticking with the proposed implementation which sets and updates `where` non-confrontationally and only when necessary.

```
if (dateFrom) {
    args.where = {
      ...args.where,
      startTime: { gte: dateFrom },
    };
  }
```

@alishaz-polymath → Do you think we can look into approving / merging this as is?

Reply:
Sure 👍 
I think we can optimize the code as a follow up. 🙌 Thanks for the PR 🎉